src/grpc/stream_infer_handler.cc

```diff
@@ -285,9 +291,13 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
           tritonserver_, shm_manager_, request, std::move(serialized_data),
           response_queue_, &state->alloc_payload_);
     }
+
+    auto request_release_payload =
```

Comment:
One thing to check is line src/grpcs/stream_infer_handler.cc#352

I found that when moving to the shared pointer / custom deleter I needed to remove any explicit TRITONSERVER_InferenceRequestDelete() calls as they would be called automatically via shared pointer cleanup. In the case that ownership is shared with request_release_payload and state - then it gets cleaned up when those are released.

If they are not shared (if some error happens before then) - then the shared pointer locally would get released. I'm not sure in this case if that is the case or not -but I would quickly check).

Its a bit of a tricky condition - do we have tests that cover memory leaks in error handling?

Reply:
Nice catch! The custom deleter would be invoked if the ownership of `request_release_payload` is held in case of InferAsync fails.